app/models/miq_expression/expression/base.rb

```diff
@@ -0,0 +1,3 @@
+class MiqExpression::Expression::Base
```

Comment:
Components! I think that's it. I'm applying the composite pattern here so I think that's the right name. (I thought SubExpression too but I didn't want to confuse that with what a sub-expressions of composites). Thank you!